### PR TITLE
Adds generator for queen attack exercise

### PR DESCRIPTION
**Pull request by samjonester**
The generator for this test uses the interface with methods `test_name`, `skipped`, `workload`. It originally comes from issue #439

### CODE REVIEW

**Comment by Insti** [comment]
Thanks @samjonester 

1. It's useful if you add a bit more information about the PR to the commit message and the first post of the pull request.

Things like: What you've added, how it differers from what was there before, if there were any special challenges/differences from the 'average' test generator.

2. The build seems to be failing due to Rubocop errors: Is there a problem with the file or is Rubocop wrong in this case?
```
3.57s$ rubocop -fs -D
== lib/queen_attack_cases.rb ==
E: 51: 38: unexpected token tCOLON
(Using Ruby 2.0 parser; configure using TargetRubyVersion parameter, under AllCops)
E: 51: 43: unexpected token tRPAREN
(Using Ruby 2.0 parser; configure using TargetRubyVersion parameter, under AllCops)
```

**Comment by samjonester** [comment]
@Insti Thanks for the feedback! I've fixed the rubocop issue, and provided some more documentation. Let me know if anything else isn't up to snuff :)

**Comment by Insti** [comment]
Great, thanks @samjonester, someone will review it properly soon, but it looks good at first-glance.

File: exercises/queen-attack/queen_attack_test.rb [hunk 1]
```diff
@@ -3,152 +3,90 @@
...
 require_relative 'queen_attack'
 
-class QueensTest < Minitest::Test
-  def test_default_positions
-    queens = Queens.new
-    assert_equal [0, 3], queens.white
-    assert_equal [7, 3], queens.black
+# Test data version:
+# 82eb00d
+class QueenTest < Minitest::Test
```
Comment (bmulvihill) [inline exercises/queen-attack/queen_attack_test.rb]: Looks like there is no space between tests in this file which could make it hard for the end user to read.

**Comment by kotp** [comment]
Thanks @samjonester

**Pull request merged**